src/services/authentication/hooks/index.js

```diff
@@ -101,6 +101,20 @@ const lowerCaseUsername = (hook) => {
 	return hook;
 };
 
+const trimUsername = (hook) => {
+	if (hook.data.username) {
+		hook.data.username = hook.data.username.trim();
+	}
+	return hook;
+};
+
+const trimPassword = (hook) => {
+	if (hook.data.password) {
+		hook.data.password = hook.data.password.trim();
```

Comment:
What if the existing password contains spaces?

Reply:
as first or last char or within?

Reply:
"  Hello World 42 !!  " is a valid password and if a user uses a password like this (that is starting/ending with spaces), this PR will prevent them from logging in, or am I getting this wrong?

Reply:
I thought of that as well. Interestingly, there already is a trim function in the hook for setting passwords. See here: https://github.com/schul-cloud/schulcloud-server/blob/ec9b3d64e4d60f731db6e910ddbc3ac1a2962159/src/services/account/hooks/index.js#L50

Reply:
Okay, I didn't know. Then we must trim here as well!